src/main/java/org/spongepowered/common/mixin/core/world/MixinGameRules.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * This file is part of Sponge, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.common.mixin.core.world;
+
+import net.minecraft.world.GameRules;
+import com.google.common.collect.ImmutableSet;
+import org.spongepowered.api.world.gamerule.DefaultGameRules;
+import org.spongepowered.asm.mixin.Final;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Shadow;
+import org.spongepowered.common.interfaces.world.IMixinGameRules;
+
+import java.lang.reflect.Field;
+import java.lang.reflect.Modifier;
+import java.util.Set;
+import java.util.TreeMap;
+
+@Mixin(GameRules.class)
+public abstract class MixinGameRules implements IMixinGameRules {
+
+    private static final Set<String> DEFAULT;
+
+    static {
+        final ImmutableSet.Builder<String> builder = ImmutableSet.builder();
+        for (Field field : DefaultGameRules.class.getFields()) {
```

Comment:
I think it would be much better to maintain a single static `GameRule` instance (which has all the default values) and check if it contains the game rule.

Basically:

```java
private static final GameRules DEFAULT_GAME_RULES = new GameRules();

// ...
if (DEFAULT_GAME_RULES.hasRule(gameRule)) {
    // ...
}
```